### PR TITLE
Improve docs and tests for `SqlInfoList

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?
Follow on to https://github.com/apache/arrow-rs/issues/4256 from @roeap 

# Rationale for this change
 While reviewing  https://github.com/apache/arrow-rs/issues/4256 from @roeap  I saw some things that could potentially be improved

# What changes are included in this PR?

Add some updated docstrings and tests

# Are there any user-facing changes?
docs

### CODE REVIEW

File: arrow-flight/src/sql/mod.rs [hunk 1]
```diff
@@ -27,6 +27,7 @@
 //! 2. Helpers for encoding and decoding FlightSQL messages: [`Any`] and [`Command`]
 //! 3. A [`FlightSqlServiceClient`] for interacting with FlightSQL servers.
 //! 4. A [`FlightSqlService`] to help building FlightSQL servers from [`FlightService`].
+//! 5. Structures to build responses for FlightSQL metadata APIs: [`SqlInfoList`]
```
Comment (alamb) [inline arrow-flight/src/sql/mod.rs:30]: Hopefully this will help finding the structure

File: arrow-flight/src/sql/sql_info.rs [hunk 2]
```diff
@@ -15,16 +15,9 @@
 // specific language governing permissions and limitations
 // under the License.
 
-//! Auxiliary module to handle [`crate::sql::CommandGetSqlInfo`] queries.
+//! [`SqlInfoList`] for building responses to [`CommandGetSqlInfo`] queries.
 //!
-//! [`crate::sql::CommandGetSqlInfo`] represents metadata requests againsts the Flight SQL server.
```
Comment (alamb) [inline arrow-flight/src/sql/sql_info.rs:20]: I moved this content to the struct itself as I think that is easier to find

File: arrow-flight/src/sql/sql_info.rs [hunk 3]
```diff
@@ -260,7 +253,37 @@ impl SqlInfoUnionBuilder {
...
+/// clients.
+///
+/// Servers construct a [`SqlInfoList`] by adding infos via
+/// [`with_sql_info`] and build the response using [`encode`].
+///
+/// The availabe configuration options are defined in the [Flight SQL protos][protos].
+///
+/// # Example
+/// ```
+/// # use arrow_flight::sql::{SqlInfoList, SqlInfo, SqlSupportedTransaction};
```
Comment (alamb) [inline arrow-flight/src/sql/sql_info.rs:269]: added doc example

File: arrow-flight/src/sql/sql_info.rs [hunk 4]
```diff
@@ -345,7 +370,53 @@ static SQL_INFO_SCHEMA: Lazy<Schema> = Lazy::new(|| {
...
+        let actual_lines: Vec<_> = formatted.trim().lines().collect();
+        assert_eq!(
+            &actual_lines, expected_lines,
+            "\n\nexpected:\n\n{:#?}\nactual:\n\n{:#?}\n\n",
+            expected_lines, actual_lines
+        );
+    }
+
+    #[test]
+    fn test_sql_infos() {
```
Comment (alamb) [inline arrow-flight/src/sql/sql_info.rs:388]: I thought this would be a useful test too

**Review by roeap — APPROVED** [review]
better tests, docs, and discoverability ❤️.

File: arrow-flight/src/sql/sql_info.rs [hunk 5]
```diff
@@ -260,7 +253,37 @@ impl SqlInfoUnionBuilder {
...
+/// A list of FlightSQL server capabilties.
+///
+/// [`CommandGetSqlInfo`] are metadata requests used by a Flight SQL
+/// server to communicate supported capabilities to Flight SQL
+/// clients.
+///
+/// Servers construct a [`SqlInfoList`] by adding infos via
+/// [`with_sql_info`] and build the response using [`encode`].
+///
+/// The availabe configuration options are defined in the [Flight SQL protos][protos].
```
Comment (viirya) [inline arrow-flight/src/sql/sql_info.rs:265]:
```suggestion
/// The available configuration options are defined in the [Flight SQL protos][protos].
```

**Review by viirya — APPROVED** [review]

**Comment by alamb** [comment]
Thanks @viirya  and @roeap

**Pull request merged**